### PR TITLE
Tweak some project settings

### PR DESCRIPTION
**Pull request by sjlongland**
* Add a `.gitignore` so that generated or temporary files are not accidentally committed.
* Run `pytest` in verbose mode so that on test failures, it dumps the full `diff` where mismatches were detected in the output.

### CODE REVIEW

File: .gitignore
Comment (diazona) [inline .gitignore:4]: Not that this is _super_ important, but I've always gone by the principle recommended in the [Git documentation](https://git-scm.com/docs/gitignore) which is that editor-specific ignore rules, or anything else that will only be relevant to some developers' workflows, should go in a local ignore file like `$HOME/.config/git/ignore` (or whatever you set `core.excludesFile` to be). So I would suggest removing these lines.

```suggestion
```

File: .gitignore
Comment (diazona) [inline .gitignore:12]:
```suggestion
# Ignore project coverage
/.coverage

# Ignore tox working directory
/.tox/
```

because we only want to ignore these directories at the root of the repo

File: .gitignore
Comment (diazona) [inline .gitignore:15]: I guess if we want to limit it to directories we could do this:

```suggestion
*.egg-info/
```

although I can't imagine any circumstances in which a _file_ matching `*.egg-info` would be part of the project so it probably doesn't matter :shrug:

File: .gitignore
Comment (diazona) [inline .gitignore:19]: This is another one that I might suggest leaving out for the same reason as the vi swap files: it's specific to each individual developer's workflow. Like, personally I use `venv` as the name when creating a local environment with `python -m venv`, or things like `dev-py36` and `dev-py310` when creating them with tox, and I'm sure anyone else who might contribute has their own naming convention for this stuff. Of course it wouldn't be hard for me to use `env/`, but I think it would be equally not hard to leave it up to each individual developer to pick their own names and add those to their local Git ignore rules, and I think that's a more scalable approach.

If anyone does commit files that shouldn't be committed, we should catch it in code review, or if there's a real concern about particular filenames we could always add a pre-commit hook to catch it as well.

**Review by diazona — COMMENTED** [review]
Thanks for getting this going! The changes I suggested to the `.gitignore` file aren't things I will 100% insist on, but I figured we should at least discuss them before proceeding to approval.

A couple other things that might be worth including in `.gitignore` are `/build/` (used by `setup.py build` and also for generated documentation) and `/dist/` (used when generating sdists and wheels), but it doesn't matter whether we add them now or in a separate PR.

File: .gitignore
Comment (sjlongland) [reply .gitignore:4]: Yep, I think a lot has changed with `git` too over the years.  I've seen this done in other places, particularly with other editors (`.idea` comes to mind with those that use IntelliJ-derived development environments) or other operating systems (`.DS_Store` on MacOS X).

I learned to use `git` back in 2008 when things were a lot more primitive, so I've probably learned bad habits here. :-)  Good to learn there is a better, less viral, way of doing things.

File: .gitignore
Comment (sjlongland) [reply .gitignore:15]: Yep, and if someone does create a `.egg-info` _file_, we don't want it here either. :-)

**Review by diazona — APPROVED** [review]

**Pull request merged**
